qualang_tools/results/results.py

```diff
@@ -98,7 +109,7 @@ def progress_counter(
     if percent:
         progress += f"{current_percent:.1f}%"
     if start_time is not None:

```

Comment:
Now you also have `self.start_time`.
So maybe if the given `start_time` is None, use the one made internally? 
And of course, write it in the docstring

Reply:
Yes but self.start_time belong to the fetching tool class and not to progress_counter which is an independent function

Reply:
You're right, my bad.
But then, why do we need `self.start_time` for the fetching tool? Is it only being used to check if it ever ran.
we can use a boolean for that:
In the init:
`self._b_first_time = True`

Reply:
This is what I did first, but then I figured that by using start_time, we can also directly get the time at which the processing starts and feed it to the progress counter so that the user does not have to import time and write start_time=time()

Reply:
Oh, so you want the users to have access to it?
In this case, I would add a getter function:
`def get_start_time():`
And make it return that time. It is not good practice to directly access the object's variables.

Reply:
See my other comment above :)

Reply:
I did the get_start_time function which is indeed better